streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -368,6 +370,7 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
      * @param queryableStoreType    accept only stores that are accepted by {@link QueryableStoreType#accepts(StateStore)}
      * @param <T>                   return type
      * @return  A facade wrapping the {@link org.apache.kafka.streams.processor.StateStore} instances
+     * @throws org.apache.kafka.streams.errors.InvalidStateStoreException if the streams are (re-)initializing
```

Comment:
@guozhangwang 

1. I don't think the check is redundant. In `QueryableStoreProvider` this is called from `KafkaStreams.store(...)` there may not exist any stores for the `name` and `type` in the streams instance - we should throw an exception here as we can't currently find the store. 

2.  We probably don't need empty check in `WrappingStoreProvider` as an exception will bubble up from `StreamThreadStateStoreProvider` if there is a rebalance.

3. I don't think we need any empty check in `CompositeReadOnlyKeyValueStore` and `CompositeReadOnlyWindowStore` as the `InvalidStateStoreException` from `StreamThreadStoreProvider` with the appropriate message will bubble up. 

Yes we update the state metadata after all the tasks and stores have been initialized